src/components/RepoList/actions.ts

```diff
@@ -2,15 +2,21 @@
 
 import { createClient } from "@/lib/supabase/server"
 import { revalidatePath } from "next/cache"
-import { redirect } from "next/navigation"
 
-export async function handleDelete(formData: FormData) {
-  "use server"
+export async function handleDelete(id: string) {
+  const { data, error } = await createClient()
+    .from("projects")
+    .delete()
+    .eq("id", id)
+    .select()
 
-  const id = formData.get("id")!
-  await createClient().from("projects").delete().eq("id", id)
   revalidatePath("/dashboard")
-  redirect("/dashboard")
+
+  if (error) {
+    throw new Error(error.message)
+  }
+
+  return data
```

Comment:
Supabse also has a `.throwOnError()` function you can chain to the query. I'd check if it makes sense on our case or not.

Reply:
### Summary:
After adding .throwOnError(), we achieved a cleaner syntax. However, an issue persists with the redirect functionality. Specifically, the redirect interrupts the execution sequence, preventing the function from returning a value or error after the redirect. Conversely, since a return statement terminates the function, we can't perform a redirect after returning a value.

### Details:
**Improved syntax:** The use of .throwOnError() has streamlined the code, making it more readable and maintainable.
Redirect issue: The challenge arises because the redirect halts further code execution. This means:
If the function redirects, it can't return a value or error afterward.
If the function returns a value or error first, it can't then redirect, since the return statement ends the function's execution.

**Current Solution:** To address this, the current approach involves invalidating the cache within the server function and handling the redirect in the client component. This separates concerns but may not be the most optimal long-term solution.

**Next Steps:** We may need to explore other strategies that allow for both returning a value/error and performing a redirect without interrupting the execution flow.